### ISSUE TITLE
Manually tweaking of package.json should be discouraged

### ISSUE THREAD

**Issue by VickyKoblinski** [survivejs/webpack-book#285]
This line below in `manuscript/developing/01_getting_started.md#setting-up-the-project` encourages editing **package.json** manually. With npm5 and **package-lock.json**, manually editing **package.json** can result in a mismatch between itself and **package-lock.json**. In some **package.json** fields it doesn't matter, but others (version, packages, maybe author), it does. I'd suggest adding a warning that manually editing **package.json** when also using **package-lock.json** (which is enabled in the config by default) isn't a best practice, *or* when instructing to `npm init -y`, also add `npm config set package-lock false` with an explanation. 

`You can tweak the generated *package.json* manually to make further changes to it even though a part of the operations modify the file automatically for you. The official documentation explains [package.json options](https://docs.npmjs.com/files/package.json) in more detail.`

https://github.com/survivejs/webpack-book/blob/dev/manuscript/developing/01_getting_started.md#setting-up-the-project

**Comment by sapegin** [comment]
I’ve never head that it’s a bad practice. And honestly I don’t know other way to do many changes. The lock file isn’t an issue because it will be updated as soon as run run `npm install` again.

**Comment by VickyKoblinski** [comment]
I see. [I was experiencing this issue recently](https://github.com/npm/npm/issues/16866) with the package-lock.json file and I believe it came from manually modifying my package.json file. After reading that issue report, I got the impression that manually editing the package.json file was the issue. I just tried recreating the issue on my home machine, but it actually worked as you described with `npm install` fixing package-lock. 

It looks like you're right, [PR #1708](https://github.com/npm/npm/pull/17508) added a change that causes package.json to overrule the package-lock if package.json is updated, and that was released as a part of npm v5.1.0.
